### PR TITLE
Wrap MOI.Complements to Knitro complementarity constraints

### PR DESCRIPTION
**Pull request by frapac**
CC https://github.com/JuliaOpt/MathOptInterface.jl/pull/913

### CODE REVIEW

**Comment by frapac** [comment]
Currently, I have tested only with a MOI version of [examples/mpec1.jl](https://github.com/JuliaOpt/KNITRO.jl/blob/master/examples/mpec1.jl) (btw, let me know if you want me to add this test to MOI). I was planning to test the linear mixed complementarity test this evening!

**Comment by odow** [comment]
> btw, let me know if you want me to add this test to MOI

Yes, please! I'm considering writing a simple MPEC solver which just does the transformation and passes off to an NLP solver.

**Comment by frapac** [comment]
@odow I made a PR directly on your development branch. I can write other tests if you think that could prove useful :)

**Comment by odow** [comment]
> I can write other tests if you think that could prove useful :)

The more the merrier.

File: src/MOI_wrapper/constraints.jl [hunk 1]
```diff
@@ -96,6 +97,32 @@ function MOI.add_constraint(model::Optimizer,
...
+        error("Bounds on variable $vi already exists.")
+    end
+    if is_fixed(model, vi)
+        error("Variable $vi is fixed. Cannot also set lower bound.")
+    end
+    lb = check_value(set.lower)
+    ub = check_value(set.upper)
+    model.variable_info[vi.value].has_lower_bound = true
+    model.variable_info[vi.value].has_upper_bound = true
+    # We assume that MOI's indexing is the same as KNITRO's indexing.
```
Comment (odow) [inline src/MOI_wrapper/constraints.jl]: What happens if variables are deleted?

File: src/MOI_wrapper/constraints.jl [hunk 2]
```diff
@@ -328,6 +356,91 @@ function MOI.add_constraint(model::Optimizer,
...
+
+# Complementarity constraints (M x + b  complements x)
+# As Knitro supports only complementarity constraints between variables,
+# we reformulate it by adding auxiliary variables:
+#
+# x_aux = Mx + b
+# (x_aux complements x)
+#
+function MOI.add_constraint(model::Optimizer,
+                            func::MOI.VectorAffineFunction, set::MOI.Complements)
```
Comment (odow) [inline src/MOI_wrapper/constraints.jl]: I don't think you need to support this. You can leave a bridge to convert `VectorAffineFunction` into `VectorOfVariables`.

File: src/MOI_wrapper/constraints.jl (hunk 1 above)
Comment (frapac) [reply src/MOI_wrapper/constraints.jl]: Currently, you cannot delete variable in a Knitro model.

File: src/MOI_wrapper/constraints.jl (hunk 2 above)
Comment (odow) [reply src/MOI_wrapper/constraints.jl]: Hmm. That sounds like a bug. Can you make a simple reproducible example?

**Pull request merged**